### PR TITLE
Allow react-native & expo to be part of devDependencies or peerDependencies

### PR DESCRIPTION
**Pull request by fabiancook**
RE: https://github.com/expo/expo-cli/issues/96

### CODE REVIEW

**Comment by andreyazevedo** [comment]
Why don't also allow `react-native` to be part of `peerDependencies`?

File: packages/xdl/src/project/Doctor.js [hunk 1]
```diff
@@ -472,9 +480,17 @@ export async function getExpoSdkStatus(projectRoot: string): Promise<ExpoSdkStat
...
-    if (pkg.dependencies['exponent']) {
+    if (pkg.dependencies && pkg.dependencies['exponent']) {
+      sdkPkg = 'exponent';
+    } else if (pkg.dependencies && pkg.dependencies['expo']) {
+      sdkPkg = 'expo';
+    } else if (pkg.devDependencies && pkg.devDependencies['exponent']) {
+      sdkPkg = 'exponent';
+    } else if (pkg.devDependencies && pkg.devDependencies['expo']) {
+      sdkPkg = 'expo';
+    } else if (pkg.peerDependencies && pkg.peerDependencies['exponent']) {
```
Comment (fson) [inline packages/xdl/src/project/Doctor.js]: No need to make this code support the legacy `exponent` dependency. None of the currently supported SDK versions were published as `exponent` anymore.

Comment (fabiancook) [reply packages/xdl/src/project/Doctor.js]: I feel like that would be a breaking change needing a major semver bump, someone may have mapped exponent manually

**Comment by fabiancook** [comment]
Hi, is it possible to get this merged in? I think removing the checks for `exponent ` should be a breaking change.

**Comment by DanCouper** [comment]
Hi, any movement on this? Just now encountering the exact issue that this PR fixes, but the PR's been sitting here for six months now, I'd rather not have to maintain a fork for a change this small.

**Comment by fabiancook** [comment]
Hi, I currently have no bandwidth to resolve the merge conflicts to close this PR.

**Comment by brentvatne** [comment]
took care of it and cleaned it up a bit with optional chaining.

thanks!

**Pull request closed without merging**